### ISSUE TITLE
Improve SyncMultiGPUTrainer to match tensorflow/benchmarks

### ISSUE THREAD

**Issue by ppwwyyxx** [ppwwyyxx/tensorpack#254]
Benchmark ResNet50 on T40.
For sync training, tensorpack runs as fast as tensorflow/benchmarks for 1,2,4 GPUs.
It becomes slower for more GPUs probably because we put a single copy of all variables on one GPU.

**Comment by ppwwyyxx** [comment]
Turned out the only change needed is to assign variables to different GPUs, and make sure sum of gradients is colocated with the variable.
Also, because variables are scattered around, BN update needs to be delayed (otherwise will hurt performance a lot).
